backend/src/supabase/functions/get-room-permission/index.ts

```diff
@@ -2,27 +2,42 @@ import { serve } from 'https://deno.land/std@0.131.0/http/server.ts';
 import { createClient } from 'https://esm.sh/@supabase/supabase-js@2';
 import { corsHeaders } from '../_shared/cors.ts';
 
+async function returnError(msgError: string){
+  return new Response(JSON.stringify({
+    error: msgError
+  }), {
+    headers: {
+      ...corsHeaders,
+      'Content-Type': 'application/json'
+    },
+    status: 400
```

Comment:
You use `returnError` when you can't find the room or the user, so you should probably return a 404 here.
Alternatively, to make it more generic, you could pass a second second parameter to this function where you specify the status code.